app/files.js

```diff
@@ -312,6 +331,58 @@ const roomArchived = {
   }
 }
 
+const userInvited = {
+  key: "userInvited",
+  noun: "User",
+  display: {
+    label: "User Joined",
+    description: "Triggers when a user invited to Room."
+  },
+  operation: {
+    inputFields: [
+      {
+        label: "Room",
+        key: "id",
+        required: true,
+        dynamic: "roomCreated.id.title"
+      },
+      {
+        label: "active",
+        key: "active",
+        type: "boolean",
+        helpText: "Return only those who are active"
+      }
+    ],
+    /**
+     * @param {ZObject} z
+     * @param {Bundle<SessionAuthenticationData, UserJoinedFields>} bundle
+     * @returns {Promise<User[]>}
+     */
+    async perform(z, bundle) {
+      /**
+       * @param {UsersList} data
+       * @param {boolean} activationStatus
+       * @returns {User[]}
+       */
+      const usersByList = function (data, activationStatus) {
+        return data
+          .filter(item => activationStatus ? item.sharedTo.activationStatus === 1 : true)
+          .map(item => item.sharedTo)
+      }
```

Comment:
I was initially confused by the fact that this is a separate function used only in one local place. It would be simpler to not create a separate function and instead use its implementation as it is.

The second one is that this function has a boolean parameter that determines whether the array of users should be filtered or not. The problem is that if this parameter is set to false, the function unnecessarily loops through the entire array. On the other hand, if it's set to true, the function checks this condition for each item in the array. It would be better to perform the filtration check before actually filtering the array.

```js
let users = []
if (bundle.inputData.active) {
  users = users.filter((item) => item.sharedTo.activationStatus === 1)
}
return users.map((item) => item.sharedTo)
```

And the last thing. It might be a good idea to move the status number to its own constant.

Reply:
> I was initially confused by the fact that this is a separate function used only in one local place. It would be simpler to not create a separate function and instead use its implementation as it is.
> 
> The second one is that this function has a boolean parameter that determines whether the array of users should be filtered or not. The problem is that if this parameter is set to false, the function unnecessarily loops through the entire array. On the other hand, if it's set to true, the function checks this condition for each item in the array. It would be better to perform the filtration check before actually filtering the array.
> 
> ```js
> let users = []
> if (bundle.inputData.active) {
>   users = users.filter((item) => item.sharedTo.activationStatus === 1)
> }
> return users.map((item) => item.sharedTo)
> ```
> 
> And the last thing. It might be a good idea to move the status number to its own constant.

Corrected.